### PR TITLE
Rework build and deployment to fix canary deploy.

### PR DESCRIPTION
**Pull request by richnew10**
This commit:

* Updates `package-lock.json`.
* Updates the Webpack invocation to correctly specify `app`.
* Updates the demo build script to use the same conditional install as the main library.
* Updates the canary deploy script to actually install dependencies as documented.

**Testing**

> 1. Have you successfully run `npm run build:release` locally?

N/A.

> 2. How did you test these changes?

These need to be exercised through the canary action.

> 3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it?

N/A.

> 4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?

No.

> 5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?

No.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/browser/package.json [hunk 1]
```diff
@@ -29,8 +29,7 @@
     "ts-loader": "^8.1.0",
     "typescript": "^4.2.3",
     "webpack": "^4.44.2",
-    "webpack-cli": "^4.6.0",
-    "webpack-dev-server": "^3.11.2"
```
Comment (devalevenkatesh) [inline demos/browser/package.json:33]: I think webpack still needs `webpack-dev-server` or it errors out. I got this when I was trying to upgrade one of the other demo. Did you ran into this?

Comment (richnew10) [reply demos/browser/package.json:33]: It works for me without it.

Comment (richnew10) [reply demos/browser/package.json:33]: Huh, it worked, and then the second invocation prompted me to install w-d-s. Weird!

File: demos/browser/package.json [hunk 2]
```diff
@@ -4,8 +4,8 @@
   "description": "Amazon Chime SDK JavaScript Demos",
   "scripts": {
     "deps": "cd ../.. && npm run build",
-    "build:fast": "tsc && webpack --config ./webpack.config.js",
-    "build": "npm run deps && npm install && npm run build:fast",
+    "build:fast": "node script/webpack-wrapper.js",
+    "build": "npm run deps && script/conditional-install && npm run build:fast",
     "start:fast": "npm run build:fast && node serve.js",
     "start:hot": "tsc && webpack-cli serve --config ./webpack.config.js",
     "start": "npm run deps && npm install && npm install aws-sdk && npm run start:fast"
```
Comment (ltrung) [inline demos/browser/package.json:11]: Don't we need to use script/conditional-install here too?
Also can we remove npm install aws-sdk? Don't know why we need to install aws-sdk again everytime.

File: demos/browser/webpack.config.js [hunk 3]
```diff
@@ -7,9 +7,11 @@ var HtmlWebpackInlineSourcePlugin = require('html-webpack-inline-source-plugin')
 var HtmlWebpackPlugin = require('html-webpack-plugin');
 /* eslint-enable */
 
-const app = process.env.npm_config_app || 'meetingV2';
-
 module.exports = env => {
+  console.info('Env:', JSON.stringify(env, null, 2));
+  console.info('App:', process.env.npm_config_app);
+  const app = env.app || process.env.npm_config_app || 'meetingV2';
+  console.info('Using app', app);
```
Comment (ltrung) [inline demos/browser/webpack.config.js:14]: Is the index `meetingV2.html` correct? Shouldn't it be correspond to the app here?

File: demos/browser/script/conditional-install [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env ruby
```
Comment (ltrung) [inline demos/browser/script/conditional-install:1]: Why cant we reuse the one in the main package?

File: script/deploy-canary-demo [hunk 5]
```diff
@@ -2,10 +2,12 @@
 set -euxo pipefail
 
 cd $GITHUB_WORKSPACE/demos/browser
+node script/conditional-install
```
Comment (ltrung) [inline script/deploy-canary-demo:5]: Doesn't npm run build also run conditional-install?

File: demos/browser/script/conditional-install (hunk 4 above)
Comment (richnew10) [reply demos/browser/script/conditional-install:1]: It's a short script and paths get confusing.

File: script/deploy-canary-demo (hunk 5 above)
Comment (richnew10) [reply script/deploy-canary-demo:5]: Ah, that's why I didn't have it in there lol

File: demos/browser/webpack.config.js (hunk 3 above)
Comment (richnew10) [reply demos/browser/webpack.config.js:14]: Unrelated bug, but yes, probably!

File: demos/browser/package.json (hunk 2 above)
Comment (richnew10) [reply demos/browser/package.json:11]: I agree :)

**Review by ltrung — APPROVED** [review]

**Pull request merged**